go/vt/schema/online_ddl.go

```diff
@@ -152,9 +152,28 @@ func ParseOnlineDDLStatement(sql string) (ddlStmt sqlparser.DDLStatement, action
 	return ddlStmt, action, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "unsupported query type: %s", sql)
 }
 
+func onlineDDLStatementSanity(sql string, ddlStmt sqlparser.DDLStatement) error {
+	// SQL statement sanity checks:
+	if !ddlStmt.IsFullyParsed() {
+		if _, err := sqlparser.ParseStrictDDL(sql); err != nil {
+			// More information about the reason why the statement is not fully parsed:
+			return vterrors.NewErrorf(vtrpcpb.Code_INVALID_ARGUMENT, vterrors.SyntaxError, "%v", err)
+		}
+		return vterrors.NewErrorf(vtrpcpb.Code_INVALID_ARGUMENT, vterrors.SyntaxError, "cannot parse statement: %v", sql)
+	}
+
+	if err := sqlparser.Walk(errorOnFKWalk, ddlStmt); err == ErrForeignKeyFound {
+		return vterrors.Errorf(vtrpcpb.Code_ABORTED, "foreign key constraint are not supported in online DDL")
```

Comment:
```suggestion
		return vterrors.Errorf(vtrpcpb.Code_ABORTED, "foreign key constraints are not supported in online DDL")
```
Also, it will be nice to link to the blog post explaining why.